wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -195,20 +195,52 @@ private fun parseMessage(path: List<Int>, helper: SourceCodeHelper, message: Des
     nestedTypes.add(parseEnum(nestedEnumPath, helper, nestedType, descs))
   }
 
+  val fieldList = message.fieldList.filter { field -> !field.hasOneofIndex() }
+  val oneOfFieldList = message.fieldList.filter { field -> field.hasOneofIndex() }
   return MessageElement(
     location = info.loc,
     name = message.name,
     documentation = info.comment,
     options = parseOptions(message.options, descs),
     reserveds = emptyList(),
-    fields = parseFields(path, helper, message.fieldList, descs),
+    fields = parseFields(path, helper, fieldList, descs),
     nestedTypes = nestedTypes,
-    oneOfs = emptyList(),
+    oneOfs = parseOneOfs(path, helper, message.oneofDeclList, oneOfFieldList, descs),
     extensions = emptyList(),
     groups = emptyList(),
   )
 }
 
+private fun parseOneOfs(
+  path: List<Int>, helper: SourceCodeHelper,
+  oneOfDeclList: List<DescriptorProtos.OneofDescriptorProto>,
+  oneOfFieldsList: List<FieldDescriptorProto>,
+  descs: DescriptorSource
+): List<OneOfElement> {
+  val info = helper.getLocation(path)
+  val result = mutableListOf<OneOfElement>()
+  val oneOfPath = mutableListOf(*path.toTypedArray())
+  oneOfPath.addAll(listOf(DescriptorProto.ONEOF_DECL_FIELD_NUMBER, 0))
+  val oneOfMap = mutableMapOf<Int, MutableList<FieldDescriptorProto>>()
+  for (field in oneOfFieldsList) {
+    val oneOfIndexList = oneOfMap[field.oneofIndex] ?: mutableListOf()
+    oneOfIndexList.add(field)
+    oneOfMap[field.oneofIndex] = oneOfIndexList
+  }
+
+  for (oneOfIndex in oneOfMap.keys) {
+    val oneOfFields = oneOfMap[oneOfIndex]!!
+    result.add(OneOfElement(
```

Comment:
no location?

Reply:
Apparently they don't have location for the OneOfElement: https://github.com/buildbreaker/wire/blob/master/wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/parser/OneOfElement.kt#L21-L27